src/util/tryOrHandleMessageError.ts

```diff
@@ -0,0 +1,18 @@
+export function tryOrHandleMessageError(target: Function, context?: any): (x?: any) => any {
+  function tryCatcher() {
+    const { target, context } = (<any> tryCatcher);
+    try {
+      target.apply(context || this, arguments);
```

Comment:
We could probably move this `context || this` to line 16. One less conditional called during each nexted return.